runtime/compiler/env/VMJ9.cpp

```diff
@@ -2309,18 +2309,39 @@ TR_MarkHotField::markHotField(J9Class * clazz, bool rootClass)
       J9ROMClass* romClass = TR::Compiler->cls.romClassOf((TR_OpaqueClassBlock*)clazz);
       J9UTF8* name = J9ROMCLASS_CLASSNAME(romClass);
       printf("%*smarked field as hot in class %.*s\n", depth(), " ", J9UTF8_LENGTH(name), J9UTF8_DATA(name));
-      }
-
-   *(UDATA *)((char *)clazz + offsetOfHotFields()) = noncoldWord | _bitValue;
-
-   return true;
+      }  
+
+      if (jitConfig->javaVM->memoryManagerFunctions->j9gc_hot_reference_field_required(jitConfig->javaVM)) {
+         if(clazz->totalInstanceSize <=512) {
```

Comment:
where does this 512 come from? it seems like a hardcoded magic number. If this is something the GC cares about extracting it to an API n the GC so the GC can adjust it without modifying the JIT code would make a lot of sense - I don't think it should be baked into the JIT

Reply:
The number 512 will be moved to a "MAX_HOT_FIELD_SIZE_IN_BYTES" constant, this was the plan all along but has not yet been done, I will do this now.